php-checks/src/main/java/org/sonar/php/checks/UnusedPrivateMethodCheck.java

```diff
@@ -53,10 +57,26 @@ public void visitClassDeclaration(ClassDeclarationTree tree) {
     }
   }
 
+  @Override
+  public void visitMethodDeclaration(MethodDeclarationTree tree) {
+    SyntaxToken firstToken = tree.modifiers().stream().findFirst().orElse(tree.functionToken());
+    firstToken.trivias().stream()
+      .flatMap(trivia -> USES_PHPDOC_PATTERN.matcher(trivia.text()).results())
+      .map(result -> result.group(2))
+      .map(methodName -> methodName.toLowerCase(Locale.ROOT))
+      .forEach(dynamicUsedMethods::add);
+
+    super.visitMethodDeclaration(tree);
+  }
+
   private void checkClass(ClassTree tree) {
     Scope classScope = context().symbolTable().getScopeFor(tree);
     for (Symbol methodSymbol : classScope.getSymbols(Kind.FUNCTION)) {
 
+      if (dynamicUsedMethods.contains(methodSymbol.name())) {
```

Comment:
I believe this condition could be moved to the `if` statement a few lines below.